token_update_nfts_transaction_e2e_test.go

```diff
@@ -0,0 +1,317 @@
+		nftSerials = append(nftSerials, receipt.SerialNumbers[0])
+	}
+
+	metadataListAfterMint := getMetadataList(t, &env, tokenID, nftSerials)
+	assert.Equal(t, initialMetadataList, metadataListAfterMint, "metadata after minting should match initial metadata")
+
+	// Update metadata for the first half of the NFTs
+	tokenUpdateNftsTransactionReceipt := updateNftMetadata(t, &env, tokenID, nftSerials[:nftCount/2], updatedMetadata, metadataKey)
+
+	// Verify updated metadata for NFTs
+	nftSerialsUpdated := tokenUpdateNftsTransactionReceipt.SerialNumbers[:nftCount/2]
+	metadataListAfterUpdate := getMetadataList(t, &env, tokenID, nftSerialsUpdated)
+	assert.Equal(t, updatedMetadataList, metadataListAfterUpdate, "updated metadata should match expected updated metadata")
+}
+
+func TestCannotUpdateNFTMetadataWhenTransactionIsNotSignedWithMetadataKey(t *testing.T) {
+	env := NewIntegrationTestEnv(t)
+
+	// Generate metadata key
+	metadataKey, err := PrivateKeyGenerateEd25519()
+	require.NoError(t, err)
+
+	nftCount := 4
+	initialMetadataList := generateNftMetadata([]byte{4, 2, 0}, nftCount)
+	updatedMetadata := []byte{6, 9}
+
+	// Create a token with a metadata key
+	tokenID := createTokenWithMetadataKey(t, &env, &metadataKey)
+
+	// Mint tokens
+	tokenMintTransactionReceipts := mintTokens(t, &env, tokenID, initialMetadataList)
+	nftSerials := make([]int64, 0)
+	for _, receipt := range tokenMintTransactionReceipts {
+		nftSerials = append(nftSerials, receipt.SerialNumbers[0])
+	}
+
+	// Assert this will fail
+	_, err = updateNftMetadataWithWrongKey(t, &env, tokenID, nftSerials, updatedMetadata, env.OperatorKey)
+	require.Error(t, err)
+}
+
+func TestCannotUpdateNFTMetadataWhenMetadataKeyWasRemoved(t *testing.T) {
+	env := NewIntegrationTestEnv(t)
+
+	// Generate metadata key
+	metadataKey, err := PrivateKeyGenerateEd25519()
+	require.NoError(t, err)
+
+	nftCount := 4
+	initialMetadataList := generateNftMetadata([]byte{4, 2, 0}, nftCount)
+	updatedMetadata := []byte{6, 9}
+
+	// Create a token with a metadata key
+	tokenID := createTokenWithMetadataKey(t, &env, &metadataKey)
+
+	// Check metadata key before removal
+	tokenInfoBeforeUpdate := getTokenInfo(t, &env, tokenID)
+	assert.Equal(t, metadataKey.PublicKey().String(), tokenInfoBeforeUpdate._ToProtobuf().MetadataKey.String(), "metadata key should match before removal")
+
+	// Remove metadata key
+	updateTokenMetadataKey(t, tokenID, nil, &env)
+
+	// Check metadata key after removal
+	tokenInfoAfterUpdate := getTokenInfo(t, &env, tokenID)
+	assert.Nil(t, tokenInfoAfterUpdate._ToProtobuf().MetadataKey, "metadata key should be nil after removal")
+
+	// Mint tokens
+	tokenMintTransactionReceipts := mintTokens(t, &env, tokenID, initialMetadataList)
+	nftSerials := make([]int64, 0)
+	for _, receipt := range tokenMintTransactionReceipts {
+		nftSerials = append(nftSerials, receipt.SerialNumbers[0])
+	}
+
+	// Check NFTs' metadata can't be updated when a metadata key is not set
```

Comment:
Isn't it actually set in:
`metadataKey, err := PrivateKeyGenerateEd25519()`

Reply:
Yeah, i mean that it's removed from the token itself, will change this comment.